### PR TITLE
Add core smoke test

### PR DESCRIPTION
**Pull request by erikzaadi**
# Description

Add a smoke test that uses the fake integration to connect to a live port instance and verify that entities arrive properly

### Content

- [x] Git ignore junit files
- [x] Update docker image labels to ensure new ghcr image permissions - according to [this Github policy](https://docs.github.com/en/packages/learn-github-packages/configuring-a-packages-access-control-and-visibility)
- [x] Add base bash port client (Will be ported to other languages)
- [x] Add smoke test that runs the fake integration in a container and assert on created entities in port
- [x] Add github action for core changes (that triggers smoke test)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: scripts/run-smoke-test.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+SCRIPT_BASE="$(cd -P "$(dirname "$0")" && pwd)"
+ROOT_DIR="$(cd -P "${SCRIPT_BASE}/../" && pwd)"
+
+RANDOM_ID="${RANDOM}${RANDOM}" # DOGE - RANDOM
+INTEGRATION_IDENTIFIER="smoke-test-integration-${RANDOM_ID}"
+BLUEPRINT_DEPARTMENT="fake-department-${RANDOM_ID}"
+BLUEPRINT_PERSON="fake-person-${RANDOM_ID}"
+BLUEPRINT_IDENTIFIERS="${BLUEPRINT_DEPARTMENT} ${BLUEPRINT_PERSON}"
+FAKE_INTEGRATION_VERSION=${FAKE_INTEGRATION_VERSION:-0.1.1-dev}
```
Comment (Tankilevitch) [inline scripts/run-smoke-test.sh:18]: what is the effect if its static dev version?

File: scripts/run-smoke-test.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+PORT_BASE_URL_FOR_DOCKER=${PORT_BASE_URL}
+
+if [[ ${PORT_BASE_URL} =~ localhost ]]; then
+    PORT_BASE_URL_FOR_DOCKER=${PORT_BASE_URL//localhost/host.docker.internal}
```
Comment (Tankilevitch) [inline scripts/run-smoke-test.sh:22]: please elaborate 🙏

File: scripts/run-smoke-test.sh [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+fi
+
+TEMP_DIR=$(mktemp -d -t smoke-test-integration.XXXXXXX)
```
Comment (Tankilevitch) [inline scripts/run-smoke-test.sh:25]: what is this for?

File: scripts/run-smoke-test.sh
Comment (Tankilevitch) [inline scripts/run-smoke-test.sh:1]: overall i feel like there are a lot of comments missing to make this readable

File: scripts/run-smoke-test.sh [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+RESOURCE_DIR_SUFFIX="integrations/fake-integration/.port/resources"
+cp -r "${ROOT_DIR}"/${RESOURCE_DIR_SUFFIX} "${TEMP_DIR}"
+
+sed -i.bak "s/fake-department/${BLUEPRINT_DEPARTMENT}/g" "${TEMP_DIR}"/resources/blueprints.json
+sed -i.bak "s/fake-person/${BLUEPRINT_PERSON}/g" "${TEMP_DIR}"/resources/blueprints.json
+sed -i.bak "s/\"fake-department\"/\"${BLUEPRINT_DEPARTMENT}\"/g" "${TEMP_DIR}"/resources/port-app-config.yml
+sed -i.bak "s/\"fake-person\"/\"${BLUEPRINT_PERSON}\"/g" "${TEMP_DIR}"/resources/port-app-config.yml
```
Comment (Tankilevitch) [inline scripts/run-smoke-test.sh:32]: should we rename the file? as its very very specific for fake-integration?

File: integrations/_infra/Dockerfile [hunk 5]
```diff
@@ -4,6 +4,7 @@ ARG BUILD_CONTEXT
 ARG INTEGRATION_VERSION
 
 LABEL INTEGRATION_VERSION=${INTEGRATION_VERSION}
+LABEL org.opencontainers.image.source https://github.com/port-labs/ocean
```
Comment (Tankilevitch) [inline integrations/_infra/Dockerfile:7]: add comment for reasoning

File: .github/workflows/smoke-test.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
+name: Smoke Test
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  detect-changes:
+    uses: ./.github/workflows/detect-changes-matrix.yml
```
Comment (Tankilevitch) [inline .github/workflows/smoke-test.yml:9]: shouldn't this be the opposite? as the smoke tests is for changes in core rather than in integrations

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: scripts/run-smoke-test.sh (hunk 1 above)
Comment (erikzaadi) [reply scripts/run-smoke-test.sh:18]: It's possible to override if wanted (will take defined env var or the default value)

File: scripts/run-smoke-test.sh (hunk 2 above)
Comment (erikzaadi) [reply scripts/run-smoke-test.sh:22]: TL;DR to make it work on docker locally while accessing port api on a local running process, you need to use 'host.docker.internal' within the container.

File: scripts/run-smoke-test.sh (hunk 3 above)
Comment (erikzaadi) [reply scripts/run-smoke-test.sh:25]: This is where I save the altered `blueprints.json` and `app-config-port.yml`

Comment (erikzaadi) [reply scripts/run-smoke-test.sh:25]: it's also mounted to the container

File: scripts/run-smoke-test.sh (hunk 4 above)
Comment (erikzaadi) [reply scripts/run-smoke-test.sh:32]: it's hard code for the fake integration at this delivery, can work on make it generic now or as a separate PR

File: .github/workflows/smoke-test.yml (hunk 6 above)
Comment (erikzaadi) [reply .github/workflows/smoke-test.yml:9]: This is me making a small AC preparation crime.
Wanted to create a convention where you'd add something according to convention to run the same smoke test for an integration later.

The detect changes also detects changes in core

File: integrations/_infra/Dockerfile (hunk 5 above)
Comment (erikzaadi) [reply integrations/_infra/Dockerfile:8]: Done

File: scripts/run-smoke-test.sh
Comment (erikzaadi) [reply scripts/run-smoke-test.sh:1]: Added some comments and tried to simplify where possible, please have a looksie

**Comment by erikzaadi** [comment]
@Tankilevitch made some more changes for readability

File: scripts/run-smoke-test.sh
Comment (erikzaadi) [reply scripts/run-smoke-test.sh:1]: `mv {bash,python}`

**Review by Tankilevitch — APPROVED** [review]
LGTM, nice smoke test ;)

**Pull request merged**